Problems when displaying in the UI manually added clusters

Hello,

I try to provision the source cluster manually, skipping the UI. 


What I do:
- I prepare and install app migration in source and target ocps.
- I "oc create -f " in the target cluster the 3 attached files in this issue , in order to provision the source cluster. A cluster, a migration cluster, and a security account secret.
[resources.zip](https://github.com/fusor/mig-ui/files/3457656/resources.zip)

Be aware that the values are modified in those resources so that I don't expose anything.

What I expect:
- The migration cluster to be "Ready" and connected.
- I expect to be able to watch the cluster in the UI too.


What it happens:
- The migration cluster is "Ready" and connected (that's OK). 
```
oc describe migclusters.migration.openshift.io  migcluster-remote 
Name:         migcluster-remote
Namespace:    mig
..............
.............
Status:
  Conditions:
    Category:              Required
    Last Transition Time:  2019-08-01T15:57:38Z
    Message:               The cluster is ready.
    Status:                True
    Type:                  Ready
Events:                    <none>

```
- But the UI fails showing 0 cluster. I get this error in the browser's console

```
chunk-e922c950.js:100 Error: Error: Error while fetching data.
    at l (sagas.ts:39)
    at l.next (<anonymous>)
    at h (redux-saga-core.esm.js:1151)
    at Object.l [as cont] (redux-saga-core.esm.js:1245)
    at b (redux-saga-core.esm.js:1008)
    at Object.t.cont (redux-saga-core.esm.js:836)
    at h (redux-saga-core.esm.js:1164)
    at l (redux-saga-core.esm.js:1245)
b @ chunk-e922c950.js:100
b @ redux-saga-core.esm.js:999
c @ redux-saga-core.esm.js:813
t.cont @ redux-saga-core.esm.js:828
h @ redux-saga-core.esm.js:1172
l @ redux-saga-core.esm.js:1245
n @ chunk-e922c950.js:166
l @ redux-saga-core.esm.js:1245
b @ redux-saga-core.esm.js:1008
c @ redux-saga-core.esm.js:813
t.cont @ redux-saga-core.esm.js:828
h @ redux-saga-core.esm.js:1172
l @ redux-saga-core.esm.js:1245
t @ redux-saga-core.esm.js:693
l @ redux-saga-core.esm.js:1245
b @ redux-saga-core.esm.js:1008
c @ redux-saga-core.esm.js:813
t.cont @ redux-saga-core.esm.js:828
h @ redux-saga-core.esm.js:1172
l @ redux-saga-core.esm.js:1245
b @ redux-saga-core.esm.js:1008
t.cont @ redux-saga-core.esm.js:836
h @ redux-saga-core.esm.js:1164
l @ redux-saga-core.esm.js:1245
Promise.then (async)
H @ redux-saga-core.esm.js:395
(anonymous) @ redux-saga-core.esm.js:1190
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
l @ redux-saga-core.esm.js:1245
(anonymous) @ redux-saga-core.esm.js:1201
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
l @ redux-saga-core.esm.js:1245
Promise.then (async)
H @ redux-saga-core.esm.js:395
(anonymous) @ redux-saga-core.esm.js:1190
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
l @ redux-saga-core.esm.js:1245
A.(anonymous function) @ redux-saga-core.esm.js:731
(anonymous) @ redux-saga-core.esm.js:1198
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
J @ redux-saga-core.esm.js:1102
A.(anonymous function) @ redux-saga-core.esm.js:533
(anonymous) @ redux-saga-core.esm.js:1198
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
J @ redux-saga-core.esm.js:1102
A.(anonymous function) @ redux-saga-core.esm.js:533
(anonymous) @ redux-saga-core.esm.js:1198
m @ redux-saga-core.esm.js:1265
(anonymous) @ redux-saga-core.esm.js:721
A.(anonymous function) @ redux-saga-core.esm.js:716
(anonymous) @ redux-saga-core.esm.js:1198
m @ redux-saga-core.esm.js:1265
h @ redux-saga-core.esm.js:1155
l @ redux-saga-core.esm.js:1245
(anonymous) @ redux-saga-core.esm.js:481
p @ redux-saga-core.esm.js:31
b @ redux-saga-core.esm.js:87
h @ redux-saga-core.esm.js:46
S.e.put @ redux-saga-core.esm.js:375
(anonymous) @ redux-saga-core.esm.js:1410
(anonymous) @ redux-logger.js:1
(anonymous) @ index.js:11
startClusterPolling @ HomeComponent.tsx:296
startDefaultClusterPolling @ HomeComponent.tsx:148
componentDidMount @ HomeComponent.tsx:164
Rc @ react-dom.production.min.js:216
Nc @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
Si @ react-dom.production.min.js:250
Li @ react-dom.production.min.js:249
ki @ react-dom.production.min.js:248
wi @ react-dom.production.min.js:245
Xc @ react-dom.production.min.js:243
enqueueSetState @ react-dom.production.min.js:130
O.setState @ react.production.min.js:13
(anonymous) @ react.js:38
Promise.then (async)
r.handlePersistorState @ react.js:37
g @ redux.js:220
d @ persistStore.js:59
C @ persistReducer.js:54
(anonymous) @ persistReducer.js:79
Promise.then (async)
(anonymous) @ persistReducer.js:78
Promise.then (async)
(anonymous) @ persistReducer.js:74
g @ redux.js:212
(anonymous) @ middleware.js:22
(anonymous) @ redux-saga-core.esm.js:1408
(anonymous) @ redux-logger.js:1
(anonymous) @ index.js:11
persist @ persistStore.js:93
j @ persistStore.js:97
t.default @ configureStore.ts:51
(anonymous) @ index.tsx:10
n @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ bootstrap:83
chunk-e922c950.js:101 The above error occurred in task l
    created by anonymous
    created by anonymous
```




Please, if you need more information, tell me and I will provide it.

Thank you very much.

--- a/src/app/cluster/duck/operations.ts
+++ b/src/app/cluster/duck/operations.ts
@@ -120,10 +120,10 @@ function groupClusters(migClusters: any[], refs: any[]): any[] {
 
     if (!mc.spec.isHostCluster) {
       fullCluster['Cluster'] = refs.find(
-        i => i.data.kind === 'Cluster' && i.data.metadata.name === mc.metadata.name
+        i => i.data.kind === 'Cluster' && i.data.metadata.name === mc.spec.clusterRef.name
       ).data;
       fullCluster['Secret'] = refs.find(
-        i => i.data.kind === 'Secret' && i.data.metadata.name === mc.metadata.name
+        i => i.data.kind === 'Secret' && i.data.metadata.name === mc.spec.serviceAccountSecretRef.name
       ).data;
     }
 

